application/controllers/survey_response.py

```diff
@@ -0,0 +1,258 @@
+        :return: Returns status code and message
+        """
+
+        tx_id = str(uuid.uuid4())
+        log.info('Adding survey response file {} for case {} with tx_id {}'.format(file, case_id, tx_id))
+
+        if case_id and file and hasattr(file, 'filename'):
+            file_name, file_extension = os.path.splitext(secure_filename(file.filename))
+            is_valid_file, msg = self._is_valid_file(file_name, file_extension)
+
+            if not is_valid_file:
+                return 400, msg
+
+            # request case_group from case service
+            case = self._get_case(case_id)
+
+            if not case or 'caseGroup' not in case:
+                return self._not_found_service_response()
+
+            case_group = case.get('caseGroup')
+            ru = case_group.get('sampleUnitRef')
+            collection_exercise_id = case_group.get('collectionExerciseId')
+
+            # request collection_exercise from service
+            collection_exercise = self._get_collection_exercise(collection_exercise_id)
+
+            if collection_exercise:
+                exercise_ref = collection_exercise.get('exerciseRef')
+                formatted_exercise_ref = self._format_exercise_ref(exercise_ref)
+                survey_id = collection_exercise.get('surveyId')
+            else:
+                return self._not_found_service_response()
+
+            # request survey service data from survey service
+            survey_ref = self._get_survey_ref(survey_id)
+
+            if not survey_ref:
+                return self._not_found_service_response()
+
+            # Create, encrypt and send message to rabbitmq
+            generated_file_name = self._generate_file_name(ru, formatted_exercise_ref, survey_ref, file_extension)
+            file_contents = file.read()
+            json_message = self._create_json_message_for_file(generated_file_name, file_contents, case_id)
+            encrypted_message = self._encrypt_message(json_message)
+
+            return self._send_message_to_rabbitmq(encrypted_message, tx_id)
+
+        else:
+            log.info('Case id or file missing')
+            return 400, INVALID_UPLOAD
+
+    @staticmethod
+    def _create_json_message_for_file(generated_file_name, file, case_id):
+        """
+          Create json message from file
+          :param generated_file_name: The generated file name
+          :param file: The file uploaded
+          :param case_id: The case UUID
+          :return: Returns json message
+        """
+
+        log.info('Creating json message')
+        file_as_string = convert_file_object_to_string_base64(file)
+
+        message_json = {
+            'filename': generated_file_name,
+            'file': file_as_string,
+            'case_id': case_id
+        }
+
+        return message_json
+
+    @staticmethod
+    def _get_case(case_id):
+        """
+        Get case details from gateway
```

Comment:
Are we still getting these details from gateway or is this just a leftover comment?

Reply:
Old comment, the gateway has been retired in collection instrument